src/py21cmfast/src/BrightnessTemperatureBox.c

```diff
@@ -1,485 +1,561 @@
-    // Do each time to avoid Python garbage collection issues
-    Broadcast_struct_global_PS(user_params,cosmo_params);
-    Broadcast_struct_global_UF(user_params,cosmo_params);
-
-    char wisdom_filename[500];
-    int i, ii, j, k, n_x, n_y, n_z;
-    float k_x, k_y, k_z;
-    double ave;
-
-    ave = 0.;
-
-    omp_set_num_threads(user_params->N_THREADS);
-
-    float *v = (float *) calloc(HII_TOT_FFT_NUM_PIXELS,sizeof(float));
-    float *vel_gradient = (float *) calloc(HII_TOT_FFT_NUM_PIXELS,sizeof(float));
-
-    float *x_pos = calloc(astro_params->N_RSD_STEPS,sizeof(float));
-    float *x_pos_offset = calloc(astro_params->N_RSD_STEPS,sizeof(float));
-    float **delta_T_RSD_LOS = (float **)calloc(user_params->N_THREADS,sizeof(float *));
-    for(i=0;i<user_params->N_THREADS;i++) {
-        delta_T_RSD_LOS[i] = (float *)calloc(user_params->HII_DIM,sizeof(float));
-    }
+    Try
+    { // Try block around whole function.
+        LOG_DEBUG("Starting Brightness Temperature calculation for redshift %f", redshift);
+        // Makes the parameter structs visible to a variety of functions/macros
+        // Do each time to avoid Python garbage collection issues
+        Broadcast_struct_global_PS(user_params, cosmo_params);
+        Broadcast_struct_global_UF(user_params, cosmo_params);
+
+        char wisdom_filename[500];
+        int i, ii, j, k, n_x, n_y, n_z;
+        float k_x, k_y, k_z;
+        float Trad_tot;
+        double ave;
 
+        ave = 0.;
 
-#pragma omp parallel shared(v,perturb_field) private(i,j,k) num_threads(user_params->N_THREADS)
-    {
+        omp_set_num_threads(user_params->N_THREADS);
+
+        float *v = (float *)calloc(HII_TOT_FFT_NUM_PIXELS, sizeof(float));
+        float *vel_gradient = (float *)calloc(HII_TOT_FFT_NUM_PIXELS, sizeof(float));
+
+        float *x_pos = calloc(astro_params->N_RSD_STEPS, sizeof(float));
+        float *x_pos_offset = calloc(astro_params->N_RSD_STEPS, sizeof(float));
+        float **delta_T_RSD_LOS = (float **)calloc(user_params->N_THREADS, sizeof(float *));
+        for (i = 0; i < user_params->N_THREADS; i++)
+        {
+            delta_T_RSD_LOS[i] = (float *)calloc(user_params->HII_DIM, sizeof(float));
+        }
+
+#pragma omp parallel shared(v, perturb_field) private(i, j, k) num_threads(user_params->N_THREADS)
+        {
 #pragma omp for
-        for (i=0; i<user_params->HII_DIM; i++){
-            for (j=0; j<user_params->HII_DIM; j++){
-                for (k=0; k<user_params->HII_DIM; k++){
-                    *((float *)v + HII_R_FFT_INDEX(i,j,k)) = perturb_field->velocity[HII_R_INDEX(i,j,k)];
+            for (i = 0; i < user_params->HII_DIM; i++)
+            {
+                for (j = 0; j < user_params->HII_DIM; j++)
+                {
+                    for (k = 0; k < user_params->HII_DIM; k++)
+                    {
+                        *((float *)v + HII_R_FFT_INDEX(i, j, k)) = perturb_field->velocity[HII_R_INDEX(i, j, k)];
+                    }
```

Comment:
Not sure I like this style here.

Reply:
Hi, I have added new FlagOptions for all the new features (USE_Radio_ACG, USE_Radio_MCG, USE_Radio_PBH, USE_Hawking_Radiation)

Reply:
FlagOptions are added for all the new features (USE_Radio_ACG, USE_Radio_MCG, USE_Radio_PBH, USE_Hawking_Radiation), defaults are set to False

Reply:
yeah I was using VS codes so the styling of BrightnessTemp.c is set by VS auto-indentation, anyway I will revert the code indentation style back to the original version

Reply:
I have reverted BrightnessTemperatureBox.c to the original indentation style